airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/write/WriteOperation.kt

```diff
@@ -23,18 +23,29 @@ import kotlinx.coroutines.runBlocking
 @Singleton
 @Requires(property = Operation.PROPERTY, value = "write")
 class WriteOperation(
-    private val inputConsumer: InputConsumer<DestinationMessage>,
+    private val inputConsumer: InputConsumer<*>,
     private val taskLauncher: TaskLauncher,
     private val taskRunner: TaskRunner
 ) : Operation {
+    val log = KotlinLogging.logger {}
+
     override fun execute() {
         runBlocking {
             launch { inputConsumer.run() }
 
             launch { taskLauncher.start() }
 
-            launch { taskRunner.run() }
+            launch {
+                try {
+                    taskRunner.run()
```

Comment:
I do not understand this at all. The double wrapping of `taskRunner.run()` inside 2 different `launch` seems very suspicious. Even the need to call the `run` twice seems odd. Clearly, I don't understand the intent or the logic of the underlying constructs

Reply:
```
try {
  launch { taskRunner.start() }
catch(t: Throwable) {
  launch { taskRunner.restart() }
  taskLauncher.handleException(t)
}
```
This is probably clearer, but I'm not 100% sure that it works as expected. I'll test it.

Reply:
I still don't understand why we need to restart the task